gitreceived/gitreceived.go

```diff
@@ -201,6 +207,22 @@ func handleChannel(conn *ssh.ServerConn, newChan ssh.NewChannel) {
 				ch.Stderr().Write([]byte("Invalid repo.\n"))
 				return
 			}
+
+			cacheKey := cmdargs[1]
+			if len(pushHook) > 0 {
+				var result bytes.Buffer
+				var errout bytes.Buffer
+				cmd := exec.Command(pushHook[0], cmdargs[0], cmdargs[1])
+				cmd.Stdout = &result
+				cmd.Stderr = &errout
+				cmd.Run()
+				if len(errout.String()) != 0 {
+					fail("pushHook", fmt.Errorf(errout.String()))
+					return
+				}
+				cacheKey = result.String()
```

Comment:
This should export an arbitrary set of env-style vars that are exposed to all subsequent hooks as described originally: https://botbot.me/freenode/flynn/2015-02-26/?msg=32869996&page=1